src/orion/storage/base.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.storage.base -- Generic Storage Protocol
+====================================================
+
+.. module:: base
+   :platform: Unix
+   :synopsis: Implement a generic protocol to allow Orion to communicate using
+   different storage backend
+
+"""
+
+from orion.core.utils import Factory
+
+
+class BaseStorageProtocol:
+    """Implement a generic protocol to allow Orion to communicate using
+    different storage backend
+
+    """
+
+    def create_experiment(self, config):
+        """Insert a new experiment inside the database"""
+        raise NotImplementedError()
+
+    def update_experiment(self, experiment, where=None, **kwargs):
+        """Update a the fields of a given trials
+
+        Parameters
+        ----------
+        experiment: Experiment
+            Experiment object to update
+
+        where: Optional[dict]
+            constraint experiment must respect
+
+        kwargs: dict
+            a dictionary of fields to update
+
+        """
```

Comment:
Maybe add a `Returns` section. This will help make the behavior coherent for all backends.